Sources/PaymentsCore/AnalyticsPayload.swift

```diff
@@ -0,0 +1,101 @@
+import UIKit
+
+struct AnalyticsPayload: Encodable {
+    
+    private let events: AnalyticsEvent
+    
+    init(eventName: String, sessionID: String) {
+        events = AnalyticsEvent(eventName: eventName, sessionID: sessionID)
+    }
+}
+
+private struct AnalyticsEvent: Encodable {
+    
+    let eventParams: AnalyticsEventParams
+    
+    init(eventName: String, sessionID: String) {
+        eventParams = AnalyticsEventParams(eventName: eventName, sessionID: sessionID)
+    }
+}
+
+private struct AnalyticsEventParams: Encodable {
+    
+    enum CodingKeys: String, CodingKey {
+        case appID = "app_id"
+        case appName = "app_name"
+        case clientSDKVersion = "c_sdk_ver"
+        case clientOS = "client_os"
+        case component = "comp"
+        case deviceManufacturer = "device_manufacturer"
+        case eventName = "event_name"
+        case eventSource = "event_source"
+        case packageManager = "ios_package_manager"
+        case isSimulator = "is_simulator"
+        case merchantAppVersion = "mapv"
+        case deviceModel = "mobile_device_model"
+        case platform = "platform"
+        case sessionID = "session_id"
+        case timestamp = "t"
+        case tenantName = "tenant_name"
+    }
+    
+    let appID: String = Bundle.main.infoDictionary?[kCFBundleIdentifierKey as String] as? String ?? ""
+    
+    let appName: String = Bundle.main.infoDictionary?[kCFBundleNameKey as String] as? String ?? ""
+
+    let clientSDKVersion = PayPalCoreConstants.paypalSDKVersion
+
+    let clientOS: String = UIDevice.current.systemName + " " + UIDevice.current.systemVersion
+
+    let component = "ppunifiedsdk"
+
+    let deviceManufacturer = "Apple"
+
+    let eventName: String
+
+    let eventSource = "mobile-native"
+
+    let packageManager: String = {
```

Comment:
General question. Do we have a preference for how we do computed properties? We use both the format here as well as this format in other places:
```
var testProperty: String {
    "my string"
}
```

Reply:
This is technically not a computed property, the property is being initialized with a closure. This is because `CodingKey` doesn't support computed variables. Sammy and I also played around with the format below for both `packageManager` and `isSimulator`, but figured the closure syntax was a little less visually cluttered.

```
#if COCOAPODS
let packageManager = "Cocoapods"
#elseif SWIFT_PACKAGE
let packageManager = "Swift Package Manager"
#else
let packageManager = "Carthage or Other"
#endif
```

@scannillo come to think of it - it may be possible to use regular computed variables if we do the custom `encode()` function 🤔

Reply:
[This](https://www.donnywals.com/using-closures-to-initialize-properties-in-swift/) is a good read. 

I don't see much benefit to us exploring using a custom `encode()` function here, just to use computed properties. Using the closure lets us use `let` and is only computed the first time it's initialized.